File: 백준/Silver/4659. 비밀번호 발음하기/비밀번호 발음하기.cc
#include <bits/stdc++.h>
using namespace std;

string s;

bool isVowel(char c) {
	return c == 'a' || c == 'e' || c == 'i' || c == 'o' || c == 'u';
}

bool check(string s) {
	bool flag = false;
	for (char c : s) {
		if (isVowel(c))
			flag = true;
	}

	if (s.size() >= 3) {
		for (int i = 0; i < s.size() - 2; i++) {
			if ((isVowel(s[i]) && isVowel(s[i + 1]) && isVowel(s[i + 2])) || (!isVowel(s[i]) && !isVowel(s[i + 1]) && !isVowel(s[i + 2])))
				flag = false;
		}
	}

	if (s.size() >= 2) {
		for (int i = 0; i < s.size() - 1; i++) {
			if (s[i] == s[i + 1] && s[i] != 'e' && s[i] != 'o')
				flag = false;
		}
	}

	return flag;		
}

int main()
{
	ios_base::sync_with_stdio(false);
	cin.tie(NULL);
	cout.tie(NULL);

	while (cin >> s) {
		if (s == "end")
			break;
		
		cout << "<" << s << "> is ";
		if (!check(s))
			cout << "not ";
		cout << "acceptable.\n";
	}

	return 0;
}

File: 백준/Silver/4659. 비밀번호 발음하기/비밀번호 발음하기.java
import java.io.*;
import java.util.*;

public class Main {
	public static void main(String[] args) throws IOException {
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));

		while (true) {
			String s = br.readLine();
			if (s.equals("end"))
				break;

			boolean hasVowel = false;
			for (char c : s.toCharArray()) {
				if (isVowel(c)) {
					hasVowel = true;
					break;
				}
			}

			boolean flag = !hasVowel;

			if (s.length() >= 2) {
				for (int i = 1; i < s.length(); i++) {
					if (s.charAt(i) == s.charAt(i - 1) && s.charAt(i) != 'e' && s.charAt(i) != 'o') {
						flag = true;
						break;
					}
				}
			}

			if (s.length() >= 3) {
				for (int i = 0; i < s.length() - 2; i++) {
					String tmp = s.substring(i, i + 3);
					char c1 = tmp.charAt(0);
					char c2 = tmp.charAt(1);
					char c3 = tmp.charAt(2);
					if ((isVowel(c1) && isVowel(c2) && isVowel(c3)) || (!isVowel(c1) && !isVowel(c2) && !isVowel(c3))) {
						flag = true;
						break;
					}
				}
			}

			String ans = flag ? "<%s> is not acceptable.\n" : "<%s> is acceptable.\n";
			bw.write(String.format(ans, s));
		}

		bw.flush();
	}

	static boolean isVowel(char c) {
		return c == 'a' || c == 'e' || c == 'i' || c == 'o' || c == 'u';
	}
}
